src/setup_payload/tests/TestHelpers.h

```diff
@@ -17,24 +17,24 @@
 
 #include <bitset>
 
-#include <setup_payload/Base38.cpp>
-#include <setup_payload/QRCodeSetupPayloadGenerator.cpp>
-#include <setup_payload/QRCodeSetupPayloadParser.cpp>
-#include <setup_payload/SetupPayload.cpp>
+#include <setup_payload/Base38.h>
+#include <setup_payload/QRCodeSetupPayloadGenerator.h>
+#include <setup_payload/QRCodeSetupPayloadParser.h>
+#include <setup_payload/SetupPayload.h>
 
 namespace chip {
 
-const uint16_t kSmallBufferSizeInBytes   = 1;
-const uint16_t kDefaultBufferSizeInBytes = 512;
+static constexpr uint16_t kSmallBufferSizeInBytes   = 1;
```

Comment:
Unlike unit tests on PC, tests are built in a single blob on target. The variable being extern was causing an ODR violation and link was failing. We can probably use `inline` instead since C++17 is used by CHIP.

Reply:
I see, it is just pigweed which uses C++17. In that case, the other option is to create a C++ file.

Reply:
Fixed - changes revert

Reply:
Isn't internal linkage here the default even in C++98?

Reply:
(this was a change from C)

> Change: A name of file scope that is explicitly declared const, and not explicitly declared extern, has
> internal linkage, while in C it would have external linkage
> Rationale: Because const objects can be used as compile-time values in C++, this feature urges programmers to provide explicit initializer values for each const. This feature allows the user to put const
> objects in header files that are included in many compilation units.
> Effect on original feature: Change to semantics of well-defined feature.
> Difficulty of converting: Semantic transformation
> How widely used: Seldom

Reply:
I see what the issue was... we were including source files. Moving away from that is a good fix, thanks.

Reply:
Fixed